### ISSUE TITLE
headline task icon ONLY shown after save

### ISSUE THREAD

**Issue by wangzhaohe** [leo-editor/leo-editor#1568]
The headline icon was shown immediately before, but now be shown after save.

Leo Log Window
Leo 6.3-devel, devel branch, build 8051ecd172
2020-04-05 05:55:36 -0500
Python 3.7.3, PyQt version 5.12.3
darwin

**Comment by vitalije** [comment]
@wangzhaohe can you give more details please. Which headline icon is not shown? Can you share a minimal Leo document that demonstrates this bug? I can't reproduce it.

**Comment by wangzhaohe** [comment]
Could the pic show my operations?

<img width="981" alt="icon" src="https://user-images.githubusercontent.com/2628013/79038424-bfa57e80-7c0b-11ea-8d9b-da85028ea9a8.png">

**Comment by wangzhaohe** [comment]
Perhaps with a gif can show clearly.

![3](https://user-images.githubusercontent.com/2628013/79040167-1c5b6600-7c19-11ea-92f1-d600fe4f4b97.gif)

**Comment by vitalije** [comment]
@wangzhaohe please test 8390038. I hope it fixes this bug.

**Comment by wangzhaohe** [comment]
I have run command `git pull`, but this bug still there. Please try again, thanks!

Leo Log Window
Leo 6.3-devel, devel branch, build 8390038749
2020-04-11 17:34:16 +0200
Python 3.7.3, PyQt version 5.12.3
darwin

**Comment by vitalije** [comment]
@wangzhaohe 
Can you write down the list of steps that you use to reproduce bug.

When I try to add icons from the Task pane, they are immediately shown.
This is what I do:

1. click on any node
2. click on any icon in the Task pane
3. icon is shown at the selected node.

**Comment by btheado** [comment]
@vitalije, I follow your steps on 8390038749 and I see the same misbehavior as @wangzhaohe (the icon doesn't appear).

```
Leo 6.3-devel, devel branch, build 8390038749
2020-04-11 17:34:16 +0200
Python 3.7.5, PyQt version 5.13.0
linux
```
I used `git bisect` and it shows 7908fddee is the first bad commit. Not sure if that is enough to help you.

**Comment by btheado** [comment]
@vitalije , one strange thing I notice about commit 7908fdd is that one of the code changes appears in the leo/docs/leoAttic.txt. Not sure if that is related to the issue.

When adding the icon to the node, the function `icon_filenames_for_node` is returning `['box08.png']`.

After running the save command, I see `icon_filenames_for_node` is called twice. The first time, the output is the same as above. The 2nd time the output is `['box00.png', '/home/btheado/src/leo-editor/leo/themes/leo_dark_0/Icons/cleo/pri1.png']`

And after running the save command, the icon appears just as @wangzhaohe observed.

**Comment by wangzhaohe** [comment]
My operational step is the follow:

1. click on any node
2. click on any icon in the Task pane
3. icon is not shown at the selected node.
4. run command `save`
5. icon is shown at the selected node

**Comment by vitalije** [comment]
Now it should be fixed in c02a2fa.
Please test and confirm that it solved the issue.

**Comment by wangzhaohe** [comment]
> Now it should be fixed in [c02a2fa](https://github.com/leo-editor/leo-editor/commit/c02a2fa62e29e434a00972ec81cb71b65b171f00).
> Please test and confirm that it solved the issue.

It runs well now! Thanks for your good job.
